pkg/action/action.go

```diff
@@ -128,12 +149,100 @@ func (a *Action) SetInput(input Input) (err error) {
 	//if err = a.ValidateInput(input); err != nil {
 	//	return err
 	//}
+
+	err = a.processArgs(input.Args)
+	if err != nil {
+		return err
+	}
+
+	err = a.processOptions(input.Opts)
+	if err != nil {
+		return err
+	}
+
 	a.input = input
 	// Reset to load the action file again with new replacements.
 	a.Reset()
 	return a.EnsureLoaded()
 }
 
+// nolint:dupl
+func (a *Action) processOptions(opts TypeOpts) error {
+	processors := a.GetProcessors()
+	for _, optDef := range a.ActionDef().Options {
+		if _, ok := opts[optDef.Name]; !ok {
+			continue
+		}
+
+		value := opts[optDef.Name]
+		toApply := optDef.Process
+		for _, processor := range toApply {
+			if processor.Processor == "" {
+				return errInvalidProcessor
+			}
+
+			proc, ok := processors[processor.Processor]
+			if !ok {
+				return fmt.Errorf(tmpNonExistProcessor, processor.Processor)
+			}
+
+			if !proc.IsApplicable(optDef.Type) {
+				return fmt.Errorf(tmpNotApplicableProcessor, optDef.Type)
+			}
+
+			newValue, err := proc.Execute(value, processor.Options)
+			if err != nil {
+				return err
+			}
+
+			value = newValue
+		}
+
+		opts[optDef.Name] = value
+	}
+
+	return nil
+}
+
+// nolint:dupl
```

Comment:
Can we move that to a function. The only difference is loop wrappers. Handling of processor is the same

Reply:
created another function